imagersite/imager_images/views.py

```diff
@@ -1,3 +1,35 @@
 from django.shortcuts import render
+from django.core.urlresolvers import reverse
+from django.views.generic.edit import CreateView
+from .models import Album, Photo, AlbumForm, PhotoForm
+from django.http import HttpResponseRedirect
 
 # Create your views here.
+
+# class AddAlbumView(CreateView):
+#     model = Album
+#     # exclude = ['date_published', 'owner']
+#     fields = ['cover', 'title', 'description', 'published']
+#     template_name = "add_album.html"]
+
+
+def add_album(request):
+    form = AlbumForm()
+    if request.method == 'POST':
+        form = AlbumForm(request.POST, request.FILES)
+        if form.is_valid():
+            form.instance.owner = request.user
+            form.save()
+            return HttpResponseRedirect(reverse('library_view'))
+    return render(request, 'add_album.html', context={'form': form})
+
+
+def add_photo(request):
+    form = PhotoForm()
+    if request.method == 'POST':
+        form = PhotoForm(request.POST, request.FILES)
+        if form.is_valid():
+            form.instance.owner = request.user
+            form.save()
+    form.fields['albums'].queryset = Album.objects.filter(owner__pk=request.user.pk)
```

Comment:
If you instantiate the form once, at the top, with `request.POST` and `request.FILES` already in place, then you can move this LOC to above the if statement too.  Then the queryset that excludes albums other than those belonging to the user would be in place for use in validation.  Otherwise, it isn't, and although I might not see someone else's albums listed in the drop-down on the page, nothing would prevent me from submitting a POST request with some random number as an album ID and injecting my photo into someone else's album.  This is a security hole.